### PR TITLE
Deploy script

### PR DESCRIPTION
**Pull request by carloscdias**
This PR make it possible to deploy the application to azure pushing to its git repository. It also makes use of environment variables to store sensitive data.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  DEPLOYMENT_SOURCE=$SCRIPT_DIR
+fi
+
+if [[ ! -n "$NEXT_MANIFEST_PATH" ]]; then
+  NEXT_MANIFEST_PATH=$ARTIFACTS/manifest
+
+  if [[ ! -n "$PREVIOUS_MANIFEST_PATH" ]]; then
+    PREVIOUS_MANIFEST_PATH=$NEXT_MANIFEST_PATH
+  fi
+fi
```
Comment (jflory7) [inline deploy.sh]: I'm guessing lines 39-45 are Azure-specific logic? I'm not sure I follow what the manifest paths are. Maybe we should leave some comments here to help us remember or to explain to someone else who needs the script in the future.

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  else
+    # In case we are running on kudu service this is the correct location of kuduSync
+    KUDU_SYNC_CMD=$APPDATA/npm/node_modules/kuduSync/bin/kuduSync
+  fi
+fi
+
+# Node Helpers
+# ------------
+
+selectNodeVersion () {
```
Comment (jflory7) [inline deploy.sh]: Are we installing for a specific version of Node? I'm not sure if I understand why this method is necessary.

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  eval $NPM_CMD install --no-optional --silent --production
+  ##################
+  echo Creating config file for build process...
+  cp "src/config-sample.js" "src/config.js"
+  ##################
+  echo Building react-app...
+  eval $NPM_CMD run build
+  #exitWithMessageOnError "npm failed"
+  cd - > /dev/null
+fi
```
Comment (jflory7) [inline deploy.sh]: This is super awesome! 🎉 💯 🎉

File: deploy.sh (hunk 1 above)
Comment (carloscdias) [reply deploy.sh]: Yes, those lines deal with Azure-specific logic. Most of them are covered here: https://github.com/projectkudu/kudu/wiki/Custom-Deployment-Script

File: deploy.sh (hunk 2 above)
Comment (carloscdias) [reply deploy.sh]: Yes, we are selecting one of the many available on the server. We can see that from the output of the deployment process too, it looks like this:

```bash
remote: Detecting node version spec...
remote: Using appsetting WEBSITE_NODE_DEFAULT_VERSION value: 8.0
remote: Node.js versions available on the platform are: 4.4.7, 4.5.0, 6.2.2, 6.6.0, 6.9.3, 6.10.3, 6.11.0, 8.0.0, 8.1.0.
remote: Resolved to version 8.0.0
remote: Detecting npm version spec...
remote: Using default for node 8.0.0: 5.0.3
remote: NPM versions available on the platform are: 2.15.8, 2.15.9, 3.9.5, 3.10.3, 3.10.10, 5.0.3.
remote: Resolved to version 5.0.3

```

**Comment by carloscdias** [comment]
@jflory7 Most part of the code was auto-generated using https://github.com/projectkudu/KuduScript following the steps mentioned here: https://github.com/projectkudu/kudu/wiki/Custom-Deployment-Script#user-content-custom-deployment-script-generator
I have added the steps necessary for our special case below the deployment section

File: deploy.sh (hunk 1 above)
Comment (jflory7) [reply deploy.sh]: @carloscdias This makes sense. We could drop this link as a comment somewhere in the script, what do you think?

File: deploy.sh (hunk 2 above)
Comment (jflory7) [reply deploy.sh]: @carloscdias Okay, I see. Maybe noting that we're targeting 8.0 is helpful here (unless this is dynamically set by an environment variable?).

File: deploy.sh (hunk 1 above)
Comment (carloscdias) [reply deploy.sh]: Doing it right now

File: deploy.sh (hunk 2 above)
Comment (carloscdias) [reply deploy.sh]: It is set dynamically within azure

**Pull request merged**